### PR TITLE
[WIP:#5] Added support for Create, Read and Delete support.

### PR DESCRIPTION
**Pull request by chandidas**
Add support for vDS portgroup creation.

### CODE REVIEW

File: vsphere/resource_vsphere_vds_portgroup.go [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/types"
+	"golang.org/x/net/context"
+)
+
+var PortgroupTypes = []string{
```
Comment (rachappag) [inline vsphere/resource_vsphere_vds_portgroup.go]: No need to expose this outside package

Comment (chandidas) [inline vsphere/resource_vsphere_vds_portgroup.go]: Done

File: vsphere/resource_vsphere_vds_portgroup.go [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+	string(types.DistributedVirtualPortgroupPortgroupTypeEarlyBinding),
+	string(types.DistributedVirtualPortgroupPortgroupTypeLateBinding),
+	string(types.DistributedVirtualPortgroupPortgroupTypeEphemeral),
+}
+
+const (
+	PortgroupVlanTypeNone     = "none"
+	PortgroupVlanTypeVlan     = "vlanid"
```
Comment (chandidas) [inline vsphere/resource_vsphere_vds_portgroup.go]: Done

File: vsphere/resource_vsphere_vds_portgroup.go [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+	PortgroupVlanTypePVid     = "pvid"
```
Comment (chandidas) [inline vsphere/resource_vsphere_vds_portgroup.go]: Done

File: vsphere/resource_vsphere_vds_portgroup.go [hunk 4]
```diff
@@ -0,0 +1,378 @@
...
+	PortgroupVlanTypeTrunking = "trunking"
+)
+
+var VlanType = []string{
```
Comment (chandidas) [inline vsphere/resource_vsphere_vds_portgroup.go]: Done

File: vsphere/resource_vsphere_vds_portgroup.go [hunk 5]
```diff
@@ -0,0 +1,378 @@
...
+
+			"portgroup_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"portgroup_type": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "earlyBinding",
```
Comment (chandidas) [inline vsphere/resource_vsphere_vds_portgroup.go]: Done

File: vsphere/resource_vsphere_vds_portgroup.go [hunk 6]
```diff
@@ -0,0 +1,378 @@
...
+
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "Created by Terraform",
+			},
+
+			"num_ports": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
```
Comment (chandidas) [inline vsphere/resource_vsphere_vds_portgroup.go]: Done

File: vsphere/resource_vsphere_vds_portgroup.go [hunk 7]
```diff
@@ -0,0 +1,378 @@
...
+				Default:  8,
+			},
+			"vlan": &schema.Schema{
+				Type:     schema.TypeList,
```
Comment (chandidas) [inline vsphere/resource_vsphere_vds_portgroup.go]: Done

File: vsphere/resource_vsphere_vds_portgroup.go [hunk 8]
```diff
@@ -0,0 +1,378 @@
...
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": &schema.Schema{
+							Type:         schema.TypeString,
+							Optional:     true,
+							Default:      "none",
```
Comment (chandidas) [inline vsphere/resource_vsphere_vds_portgroup.go]: Done

File: vsphere/resource_vsphere_vds_portgroup.go [hunk 9]
```diff
@@ -178,111 +186,167 @@ func resourceVSphereVdPortgroupCreate(d *schema.ResourceData, meta interface{})
...
+	dcName := d.Get("datacenter").(string)
+	pgName := d.Get("portgroup_name").(string)
 
-	pg, err := object.NewSearchIndex(client.Client).FindByInventoryPath(
-		context.TODO(), d.Id())
+	log.Printf("[INFO] reading vDS portgroup: [%s]", d.Id())
 
+	netRef, err := findNetObjectByName(dcName, pgName, client)
 	if err != nil {
+		d.SetId("")
```
Comment (rachappag) [inline vsphere/resource_vsphere_vds_portgroup.go]: Do we need to set the ID to empty string when find net object operation fails ?

**Pull request merged**